### PR TITLE
CWL: Pass local temporary directory to jobs

### PR DESCRIPTION
**Pull request by chapmanb**
This fixes problems on shared filesystem batch scheduler systems which
do not automatically transfer TMPDIR from the parent. Failing to do this
results in jobs run on non-shared local temporary directories which are
not accessible to the parent fileStorage manager. This avoids the issue
by explicitly setting and passing a temporary directory in the defined
outdir. Required for distributed runs on SGE clusters so supplements #1105
with the necessary CWL specific fixes.

### CODE REVIEW

**Comment by benedictpaten** [comment]
jenkins, test this please.

**Comment by cket** [comment]
jenkins, test this please.

**Comment by cket** [comment]
@benedictpaten it looks like Jenkins' admins are separate from Github, Jenkins seems to be ignoring me.
The integration test failures seem unrelated so hopefully they'll pass on the next run.

**Comment by benedictpaten** [comment]
jenkins, test this please

File: src/toil/cwl/cwltoil.py [hunk 1]
```diff
@@ -245,12 +245,15 @@ def run(self, fileStore):
         adjustFilesWithSecondary(cwljob, functools.partial(getFile, fileStore, inpdir, index=index))
 
         # Run the tool
+        opts = copy.deepcopy(self.executor_options)
+        # Exports temporary directory for batch systems that reset TMPDIR
+        os.environ["TMPDIR"] = os.path.realpath(opts.pop("tmpdir", None) or tmpdir)
```
Comment (hannes-ucsc) [inline src/toil/cwl/cwltoil.py]: tempfile.gettempdir would have been more portable.

**Pull request merged**